src/main/scala/com/microsoft/azure/iot/iothubreact/checkpointing/OffsetLoader.scala

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.azure.iot.iothubreact.checkpointing
+
+import akka.util.Timeout
+import com.microsoft.azure.iot.iothubreact.{Logger, Retry}
+import com.microsoft.azure.iot.iothubreact.checkpointing.CheckpointService.GetOffset
+import com.microsoft.azure.iot.iothubreact.config.IConfiguration
+import com.microsoft.azure.iot.iothubreact.scaladsl.IoTHubPartition
+
+import scala.concurrent.Await
+
+object OffsetLoader extends Logger {
+
+  /** Get the offset saved for the current partition
+    *
+    * @return Offset
+    */
+  private[iothubreact] def GetSavedOffset(partition: Int, config: IConfiguration): Option[String] = {
+    import scala.language.postfixOps
+    import scala.concurrent.duration._
+    import akka.pattern.ask
+    val partitionCp = CheckpointActorSystem(config.checkpointing).getCheckpointService(partition)
+    implicit val rwTimeout = Timeout(config.checkpointing.checkpointRWTimeout)
+    try {
+      Retry(3, 5 seconds) {
+        log.debug("Loading the stream offset for partition {}", partition)
+        val future = (partitionCp ? GetOffset).mapTo[String]
+        val offset = Await.result(future, rwTimeout.duration)
+        if (offset != IoTHubPartition.OffsetCheckpointNotFound) Some(offset) else None
+      }
+    } catch {
+      case e: java.util.concurrent.TimeoutException ⇒
+        log.error(e, "Timeout while retrieving the offset from the storage")
+        throw e
+      case e: Exception ⇒
+        log.error(e, e.getMessage)
+        throw e
+    }
+  }
+
+  private[iothubreact] def GetSavedOffsets(config: IConfiguration): Map[Int, String] = {
+    (0 to config.connect.iotHubPartitions).flatMap { p ⇒
```

Comment:
If I understand your question properly, the flatMap should take care of any missing values. I just included a test to prove it.